frontend/app/src/pages/home/join.tsx

```diff
@@ -1,11 +1,40 @@
+import { useState } from "react";
 import { Page } from "@/components";
 import { FindingMatchCard, SelectPreferencesCard } from "@/features/matching";
+import { io } from "socket.io-client";
+import { useNavigate } from "react-router-dom";
+import { Preferences } from "@/types/room";
 
 function JoinPage() {
+  const navigate = useNavigate();
+  const socket = io("http://localhost:8004");
```

Comment:
can abstract into an env var?
so during deployment, pulumi will take the matching service output domain URL and pass it to the frontend as an env var